File: src/client/src/app/core/models/container.ts
import { Image } from './image';

export interface Container {
  id: string;
  name: string;
  status: string | any;
  short_id: string;
  labels: object;
  image: Image;
}

export interface RunContainerData {
  name?: string;
  image?: string;
  command?: string;
  entrypoint?: string;
  ports?: object;
  firstPort?: string;
  secondPort?: string;
}


export interface ContainerLogsData {
  logs: string;
  container: string;
}

File: src/server/schemas/container.py
import typing as t
from pydantic import BaseModel


class ContainerOptions(BaseModel):
    image: str
    name: t.Optional[str]
    ports: t.Optional[dict]
    command: t.Optional[t.Union[str, list]]
    labels: t.Optional[t.Union[dict, list]]
    environment: t.Optional[t.Union[dict, list]]
